sonar-pmd-lib/src/test/java/org/sonar/plugins/pmd/rule/RulesDefinitionXmlLoaderTest.java

```diff
@@ -294,6 +294,33 @@ public void test_deprecated_remediation_function() {
         assertThat(function.baseEffort()).isEqualTo("5min");
     }
 
+    @Test
+    public void test_filter_nop_sonar_tags() {
+        String xml = "" +
+                "<rules>" +
+                "<rule >" +
+                "<key>Rule00</key>" +
+                "<name>Rule 00</name>" +
+                "<description>Description of rule 00</description>" +
+                "<type>BUG</type>" +
+                "<tag>tests</tag>" +
+                "<tag>main-sources</tag>" +
+                "</rule>" +
+                "</rules>";
+
+        RulesDefinition.Repository rulesRepo = load(xml);
+        RulesDefinition.Rule rule0 = rulesRepo.rule("Rule00");
+
+        assertThat(rule0.scope())
+                .withFailMessage("main-sources and tests tags means scope all")
+                .isEqualTo(RuleScope.ALL);
+
+        assertThat(rule0.tags())
+                .withFailMessage("Should not contain main-sources")
+                .containsOnly("tests");
```

Comment:
rule0 does not have tag tests!
How come it doesn't fail?
Take a rule that has both tags (as input).